aspnetcore/security/data-protection/configuration/samples/9/deleteKeys/Program.cs

```diff
@@ -0,0 +1,23 @@
+using Microsoft.AspNetCore.DataProtection.KeyManagement;
+
+var builder = WebApplication.CreateBuilder(args);
+
+var app = builder.Build();
+
+app.MapDelete("/api/keys/cleanup-old-keys", (IServiceProvider services) =>
```

Comment:
Just to be explicit, exposing an endpoint for this would be an extremely bad idea.

Reply:
The expected use case is in an out-of-band tool with no internet exposure.  One can imagine building an administration interface with this sort of endpoint, but I'd want to be very clear in the sample that this would only be accessible within a cluster or behind auth (at least).

Reply:
> The expected use case is in an out-of-band tool with no internet exposure. One can imagine building an administration interface with this sort of endpoint, but I'd want to be very clear in the sample that this would only be accessible within a cluster or behind auth (at least).

I deleted the project, the surrounding code (the endpoint). We like having samples that compile for obvious validation and maintenance reasons. In this case a snippet is justified. But I'm thinking dev's who need this can figure it out without a sample. We do know customers copy/paste code we right, and that seems to be a bigger risk.

Should I just delete the remaining code>

Reply:
I can appreciate the desire to have compiling code.  One way to accomplish that would be to write a console app that connects to an existing key repository and then deletes some keys.  Alternatively, I think the fragment is a reasonable compromise.